champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/sword/Bullseye.java

```diff
@@ -216,17 +227,35 @@ public void onDamage(CustomDamageEvent event) {
     private void focusTarget(BullsEyeData playerBullsEyeData) {
         Player caster = playerBullsEyeData.getCaster();
         RayTraceResult result = caster.rayTraceEntities(64);
-        if (result == null || result.getHitEntity() == null) return;
-        if (!(result.getHitEntity() instanceof LivingEntity)) return;
-
-        if (!playerBullsEyeData.hasTarget()) {
-            playerBullsEyeData.setTarget((LivingEntity) result.getHitEntity());
-            playerBullsEyeData.setTargetFocused(new ChargeData((float)(0.35)));
-            bullsEyeData.put(caster.getUniqueId(), playerBullsEyeData);
+        if (result != null && result.getHitEntity() != null) {
```

Comment:
Instead of checking result.getHitEntity() != null, you can just do this instead

```
if(result != null && !(result.getHitEntity() instanceof LivingEntity target)) {
           if (!playerBullsEyeData.hasTarget()) {
                playerBullsEyeData.setTarget(target);
                playerBullsEyeData.setTargetFocused(new ChargeData((float) (0.5)));
                bullsEyeData.put(caster.getUniqueId(), playerBullsEyeData);
                playerBullsEyeData.getTargetFocused().tick();
                playerBullsEyeData.getTargetFocused().tickSound(caster);
                playerBullsEyeData.updateColor();
            }
}
```

Reply:
The code you suggested doesnt work
![image](https://github.com/Mykindos/BetterPvP/assets/95445240/2bd597d9-b2ff-4681-8d3d-b7d5432af6ac)

Reply:
Try compile it anyway, theres no reason that shouldnt work. You can try set your IDE language level to java 21

Reply:
isn't the `(result.getHitEntity() instanceof LivingEntity target)` being negated, so if getHitEntity is an instance of LivingEntity, the code will ever run (therefore there will never be a `target`). It doesn't look like it should be being negated at all.

Reply:
We do this all over the place

![image](https://github.com/Mykindos/BetterPvP/assets/5389944/17778a92-99e9-40a9-b72f-c84e74e53c98)

Only reason I can think of it not working is his IDE thinking that the variable wasnt created

However it does raise a different question, this code is checking that the target is not a living entity, when really you want to check if it IS a living entity. Code should still compile either way though

Reply:
but it's not returning before, this code relies on that boolean statement being true, if it is true, as written, target doesn't exist.

I think you introduced a logic error, if you remove the ! it should work.

Reply:
Ye you right